api/handlers.go

```diff
@@ -28,50 +44,64 @@ func (app *Application) AllArticle(w http.ResponseWriter, r *http.Request) {
 	// Retrieve the list of articles from the database
 	articles, _ := app.DB.AllArticles()
 
-	// Prepare the response map
-	response := map[string]interface{}{
-		"status":  200,
-		"message": "Success",
-		"data":    []map[string]interface{}{},
-	}
+	// Create the response struct
+	var response models.Response
+
+	response.Status = http.StatusOK
+	response.Message = "Success"
 
-	// Check if there are articles, then add them to the response map
-	if len(articles) > 0 {
-		for _, article := range articles {
+	var dataList []models.Articles // Slice to hold the articles
 
-			response["data"] = append(response["data"].([]map[string]interface{}), map[string]interface{}{"id": article.ID, "title": article.Title, "content": article.Content, "author": article.Author})
+	// Check if there are articles, then add them to the response
+	for _, article := range articles {
```

Comment:
unnecessary iterating articles data. As seen here, only appending articles and no other process happen. service logic should return the required articles with the type of []models.Articles